### DISCUSSION TITLE
I use flowbite react's navbar but when I use my mobile phone to move the page from /a to /b the navbar does not close automatically. How to solve the problem?

### DISCUSSION THREAD

**Discussion by sxadxsx** [themesberg/flowbite-react#1162] [General]
I use flowbite react's navbar but when I use my mobile phone to move the page from /a to /b the navbar does not close automatically. How to solve the problem?

**Comment by tulup-conner** [comment]
You can track `window.location` with a `useState` and set `<Sidebar collapsed>`. It's up to you to create a React Context to achieve that.

**Reply by phoskee** [reply]
I've done [some tests](https://github.com/themesberg/flowbite-react/discussions/1131#discussioncomment-7904874) but it seems to buggy the menu button, do you have any suggestions?

**Reply by tulup-conner** [reply]
Personally I've configured this in a setup where you have a button outside the `<Sidebar>` that can toggle `<Sidebar hidden>`, which works great. If you could provide a sample repo/sandbox with your solution it would be helpful.

**Reply by phoskee** [reply]
Sure! [CodeSandbox](https://codesandbox.io/p/devbox/focused-ace-s26rch?file=%2Fapp%2Froot.tsx%3A14%2C3&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqevggg800092a675qvhfmru%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqevggg700022a67ht3dzewu%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqevggg800062a67z87q48vt%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqevggg800082a673gea47ee%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B51.54279130715%252C48.45720869285%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqevggg700022a67ht3dzewu%2522%253A%257B%2522id%2522%253A%2522clqevggg700022a67ht3dzewu%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clqevggg800082a673gea47ee%2522%253A%257B%2522id%2522%253A%2522clqevggg800082a673gea47ee%2522%252C%2522activeTabId%2522%253A%2522clqewhpff01302a675pw2prd1%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clqewhpff01302a675pw2prd1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252FPage1%2522%257D%255D%257D%252C%2522clqevggg800062a67z87q48vt%2522%253A%257B%2522id%2522%253A%2522clqevggg800062a67z87q48vt%2522%252C%2522activeTabId%2522%253A%2522clqewevnc00hx2a67jk3k1awe%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clqewevnc00hx2a67jk3k1awe%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
